src/Estimators/src/BaseEstimatorFromFootIMU.cpp

```diff
@@ -0,0 +1,293 @@
+    m_frame_H_link = BipedalLocomotion::Conversions::toManifPose(frame_H_link);
+
+    m_isInitialized = true;
+
+    return ok;
+}
+
+bool BaseEstimatorFromFootIMU::setInput(const BaseEstimatorFromFootIMUInput& input)
+{
+    m_input = input; // TO DO: check if this line goes wrong
+    return true;
+}
+
+bool BaseEstimatorFromFootIMU::advance()
+{
+    constexpr auto logPrefix = "[BaseEstimatorFromFootIMU::advance]";
+
+    m_isOutputValid = false;
+
+    if (!m_isInitialized)
+    {
+        log()->error("{} The estimator is not initialized.", logPrefix);
+        return false;
+    }
+
+    // `desiredFootPose` is intended as the output of the footstep planner.
+
+    // extracting the position part of the `desiredFootPose`.
+    Eigen::Vector3d desiredTranslation = m_input.desiredFootPose.translation();
+
+    // extracting the orientation part of the `desiredFootPose` and expressing it
+    // through RPY Euler angles.
+    Eigen::Matrix3d desiredRotation = m_input.desiredFootPose.rotation();
+    Eigen::Vector3d desiredRPY = EulerAngles_ZYX(desiredRotation);
+
+    // converting the measured foot orientation manif::SO3d --> Eigen::Matrix3d.
+    Eigen::Matrix3d measuredRotation = m_input.measuredRotation.rotation();
+    // expressing this orientation through RPY Euler angles.
+    Eigen::Vector3d measuredRPY = EulerAngles_ZYX(measuredRotation);
+
+    // desired Yaw is used instead of measured Yaw.
+    measuredRPY(2) = desiredRPY(2);
+
+    // manif::SO3d rotation matrix that employs: measured Roll, measured Pitch,
+    // desired Yaw.
+    m_measuredRotationCorrected = manif::SO3d(measuredRPY(0), measuredRPY(1), measuredRPY(2));
+
+    // clang-format off
+    //m_measuredRotationCorrected <<  cos(measuredRPY(0)) * cos(measuredRPY(1)),  cos(measuredRPY(0)) * sin(measuredRPY(1)) * sin(measuredRPY(2)) - sin(measuredRPY(0)) * cos(measuredRPY(2)), cos(measuredRPY(0)) * sin(measuredRPY(1)) * cos(measuredRPY(2)) + sin(measuredRPY(0)) * sin(measuredRPY(2)),
+    //                                sin(measuredRPY(0)) * cos(measuredRPY(1)),  sin(measuredRPY(0)) * sin(measuredRPY(1)) * sin(measuredRPY(2)) + cos(measuredRPY(0)) * cos(measuredRPY(2)), sin(measuredRPY(0)) * sin(measuredRPY(1)) * cos(measuredRPY(2)) - cos(measuredRPY(0)) * sin(measuredRPY(2)),
+    //                               -sin(measuredRPY(1)),                        cos(measuredRPY(1)) * sin(measuredRPY(2)),                                                                   cos(measuredRPY(1)) * cos(measuredRPY(2));
+    // clang-format on
+
+    // manif::SE3d pose matrix that employs: desired Position, measured Roll,
+    // measured Pitch, desired Yaw.
+    manif::SE3d T_foot_raw(desiredTranslation, m_measuredRotationCorrected);
+
+    // finding the positions of the foot corners in world frame given `T_foot_raw`
+    // pose matrix.
+    std::vector<Eigen::Vector3d> transformedFootCorners;
```

Comment:
Can you please remove the commented code and try to add the intermediate variables as attributes of the class? For instance the std::vector<Eigen::Vector3d>

Reply:
commented code removed.

Reply:
added intermediate variables as class attributes